src/Generator/CreditMemoPdfFileGenerator.php

```diff
@@ -74,21 +53,41 @@ public function generate(string $creditMemoId): CreditMemoPdf
 
         $filename = str_replace('/', '_', $number) . self::FILE_EXTENSION;
 
-        $pdf = $this->pdfGenerator->getOutputFromHtml(
-            $this->twig->render($this->template, [
-                'creditMemo' => $creditMemo,
-                'creditMemoLogoPath' => $this->fileLocator->locate($this->creditMemoLogoPath),
-            ]),
-            $this->pdfOptionsGenerator ? $this->pdfOptionsGenerator->generate() : []
-        );
+        $templateParams = [
+            'creditMemo' => $creditMemo,
+            'creditMemoLogoPath' => $this->fileLocator->locate($this->creditMemoLogoPath),
+        ];
+
+        if (null !== $this->twigToPdfGenerator) {
+            $pdf = $this->twigToPdfGenerator->generate($this->template, $templateParams);
+        } elseif (null !== $this->pdfGenerator && null !== $this->twig) {
+            $pdf = $this->pdfGenerator->getOutputFromHtml(
+                $this->twig->render($this->template, $templateParams),
+                $this->pdfOptionsGenerator ? $this->pdfOptionsGenerator->generate() : []
+            );
+        } else {
+            throw new \LogicException(sprintf('You must pass at least $twigToPdfGenerator to %s constructor.', self::class));
+        }
```

Comment:
Extract this logic to a separate method to be able to use early returns and avoid using `else`